libr/core/anal_tp.c

```diff
@@ -53,137 +93,43 @@ static void type_match(RCore *core, ut64 addr, char *name) {
-						goto out_function;
-					}
-					ut64 write_addr = sdb_num_get (trace, sdb_fmt ("%d.mem.write", j), 0);
-					if (write_addr == sp + size) {
-						ut64 instr_addr = sdb_num_get (trace, sdb_fmt ("%d.addr", j), 0);
-						r_meta_set_string (core->anal, R_META_TYPE_COMMENT, instr_addr,
-								sdb_fmt ("%s%s%s", type, r_str_endswith (type, "*") ? "" : " ", name));
-						char *tmp = sdb_fmt ("%d.mem.read", j);
-						int i2, array_size = sdb_array_size (trace, tmp);
-						for (i2 = 0; i2 < array_size; i2++) {
-							if (bp_name) {
-								int bp_idx = sdb_array_get_num (trace, tmp, i2, 0) - bp;
-								if ((v = r_anal_var_get (anal, addr, R_ANAL_VAR_KIND_BPV, 1, bp_idx))) {
-									r_anal_var_retype (anal, addr, 1, bp_idx, R_ANAL_VAR_KIND_BPV, type, -1, v->name);
-									r_anal_var_free (v);
-								}
-							}
-							int sp_idx = sdb_array_get_num (trace, tmp, i2, 0) - sp;
-							if ((v =r_anal_var_get (anal, addr, R_ANAL_VAR_KIND_SPV, 1, sp_idx))) {
-								r_anal_var_retype (anal, addr, 1, sp_idx, R_ANAL_VAR_KIND_SPV, type, -1, v->name);
-								r_anal_var_free (v);
-							}
-						}
-						break;
-					}
-
-				}
-				size += r_anal_type_get_size (anal, type) / 8;
-			}
-			break;
-		} else {
-			// type_match_reg ();
-			for (j = idx; j >= 0; j--) {
-				if (r_cons_is_breaked ()) {
-					goto out_function;
-				}
-				if (sdb_array_contains (trace, sdb_fmt ("%d.reg.write", j), place, 0)) {
-					ut64 instr_addr = sdb_num_get (trace, sdb_fmt ("%d.addr", j), 0);
-					r_meta_set_string (core->anal, R_META_TYPE_COMMENT, instr_addr,
-						sdb_fmt ("%s %s", type, name));
-					char *tmp = sdb_fmt ("%d.mem.read", j);
-					int i2, array_size = sdb_array_size (trace, tmp);
-					for (i2 = 0; i2 < array_size; i2++) {
-						if (r_cons_is_breaked ()) {
-							goto out_function;
-						}
-						if (bp_name) {
-							int bp_idx = sdb_array_get_num (trace, tmp, i2, 0) - bp;
-							if ((v = r_anal_var_get (anal, addr, R_ANAL_VAR_KIND_BPV, 1, bp_idx))) {
-								r_anal_var_retype (anal, addr, 1, bp_idx, R_ANAL_VAR_KIND_BPV, type, -1, v->name);
-								r_anal_var_free (v);
-							}
-						}
-						int sp_idx = sdb_array_get_num (trace, tmp, i2, 0) - sp;
-						if ((v = r_anal_var_get (anal, addr, R_ANAL_VAR_KIND_SPV, 1, sp_idx))) {
-							r_anal_var_retype (anal, addr, 1, sp_idx, R_ANAL_VAR_KIND_SPV, type, -1, v->name);
-							r_anal_var_free (v);
-						}
-					}
-					break;
+		for (j = idx; j >= prev_idx; j--) {
+			if (type_pos_hit (anal, trace, in_stack, j, size, place)) {
+				ut64 instr_addr = sdb_num_get (trace, sdb_fmt ("%d.addr", j), 0);
+				r_meta_set_string (anal, R_META_TYPE_COMMENT, instr_addr,
+						sdb_fmt ("%s%s%s", type, r_str_endswith (type, "*") ? "" : " ", name));
+				if (strncmp (type, "int", 3)) {
+					// change type only if not int
+					type_match_var (anal, trace, addr, type , j);
 				}
+				break;
 			}
 		}
-		free (type);
+		size += r_anal_type_get_size (anal, type) / 8;
```

Comment:
I think we should rename this function into `r_anal_type_get_bitsize` because `..get_size` implies the size in bytes, like `sizeof()` does.

Reply:
@XVilka  yup this might create confusion , i changed the name to r_anal_type_get_bitsize :+1: